garcon/activity.py

```diff
@@ -253,7 +253,16 @@ def run(self):
         previous activity is consumed (context).
         """
 
-        activity_task = self.poll()
+        try:
+            activity_task = self.poll()
+        except Exception as error:
```

Comment:
The challenge is any Exception has the potential to kill the thread & the whole worker daemon becomes a zombie at that point. It's nice being able to trigger the `on_exception` at this point since then a workflow can raise some sort of notification (like a Sentry alert) that action needs to be taken. We only noticed these exceptions when logging onto the EB running the daemon itself and looking at stderror, and I worry some other Exceptions might get hidden from us in a similar way if we aren't aggressive here.

I was going to catch the boto.exception specifically when adding the throttle sleep.